cli/tests/common.rs

```diff
@@ -228,10 +227,9 @@ pub async fn create_project() -> &'static str {
     // Attempt to create the project, ignoring conflicts.
     let mut api = PhylumApi::new(config, None).await.unwrap();
     match api.create_project(PROJECT_NAME, None).await {
-        Ok(_) | Err(PhylumApiError::Response(ResponseError { code: StatusCode::CONFLICT, .. })) => {
-        },
+        Ok(_) | Err(PhylumApiError::Other(_)) => (),
         err @ Err(_) => {
-            err.unwrap();
+            let _ = err.unwrap();
         },
```

Comment:
The code change here points out a big issue with this approach. The caller of `api.create_project()` can no longer distinguish a conflict from other errors. (Not easily at least. It's technically possible with a `downcast_ref()` and a second `match`).

There are a few ways that we could change this to allow a caller to continue to detect these errors:
1. We can add new variants to `PhylumApiError` for `ProjectAlreadyExists` and `GroupAlreadyExists` errors.
2. Instead of using anyhow context to improve the error message, we can just overwrite the `details` field in the `ResponseError` and continue to use the `PhylumApiError::Response` variant.

Of course, we could also accept that callers shouldn't be able to detect these cases and that the `PhylumApiError` type shouldn't exist (because we should be using `anyhow::Error` instead). But I'm not sure we're ready to take that step (or if that actually make sense).

I am leaning towards option 2 above since it is the simplest way to improve the error message and maintains the usage of `PhylumApiError::Response`. As an added bonus, it's also the easiest change to rip out later if the error message is improved on the API side.

Reply:
I don't like overwriting details tbh, since that means we actually lose information.

Considering we always use anyhow for everything anyway, I don't think matching on errors is a big priority in the CLI anyway? I'm not sure I'd call that a good thing, but it's pretty consistent across our entire codebase anyway.

If we decide this shouldn't be a thing, we should probably nuke the anyhow enum variant completely?

Reply:
> I don't like overwriting details tbh, since that means we actually lose information.

Yes and no. The reason the code exists is to replace a poor error message with a better one. So while we do technically lose information, it's just the poor error message that is lost, which is expected.

Reply:
It's not poor in general though, it's just poor at the toplevel. As a detail it still might be interesting.